docs/RFCS/certificate_rotation.md

```diff
@@ -0,0 +1,293 @@
+- Feature Name: certificate rotation
+- Status: draft
+- Start Date: 2017-03-18
+- Authors: @mberhault
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: [1674](https://github.com/cockroachdb/cockroach/issues/1674)
+
+# Summary
+
+This RFC proposes changes to certificate and key management.
+
+The main goals of this RFC are:
+* support addition and use of new certs without node restart
+* decouple CA from node certificate rotation
+* simplify generation and use of certificates and keys
+
+Out of scope for this RFC:
+* certificate revocation
+* certificate/key deployment
+* automatic certificate renewal
+
+# Motivation
+
+Anytime certificates and keys are used, we must allow for rotation,
+either due to security concerns or standard expiration.
+
+This RFC is concerned with rotation due to expiration, meaning we do not
+consider certificate revocation.
+
+Our current certificate use allows for a single CA certificate  and a single
+server certificate/private key pair, with node restarts being required to
+update any of them.
+
+We wish to be able to push new certificates to nodes and use them
+without restart or connection termination.
+
```

Comment:
Can you clarify here or later what you think should happen with client (not node) certificates.